### ISSUE TITLE
Licenses for iOS App

### ISSUE THREAD

**Issue by margitbomholt** [Esri/runtime-questions#24]
If we use the runtime parts that requires licensing, e.g local editing, and we dont want to use named Arcgis Online users, then we buy licenses in bundles of e.g. 50. But if we distribute the app on AppStore, mulitple users Can download the app. We provide loginin the App, to ensure that it will only be uset by our Own customers. 
How will Esri register / count the nummer of users, and there by the required number of licenses?

**Comment by madmapper** [comment]
Very interested in this as well - might it be possible to adopt something like the Unreal engine (5% of revenue over $3k) or Unity ($1500 or $75/mo)

**Comment by Umbrah** [comment]
iOS is, because of the small market share and development cost not really interesting for us, but for web/Android and other platforms it's actually something we deal with: our company, Ordina GEO-ICT, has built mass-data-gathering apps (mainly for events) in the past, and it's certainly something we intend to continue doing: people download an app, and collect some simple data, but without a login/GIS login/named user. An example would be an app such as the ESRI Events app with added functionality or the Amsterdam Canal Pride which can have hundreds of thousands of visitors.

Long story short: we'd love to pay either per app (and it's stored to our on-premise installs), or per functionality. At the moment, if we want to embed maps in "offline" apps with limited tracking/editing functionality we're pretty much forced to go Google/OSMDroid API's or a custom "hack": mass volume apps are a huge demand and named users are a no-go at that point.

**Comment by madmapper** [comment]
Can't say I agree about the small market share or high dev cost of iOS :-) - but other than that, spot on!  Our apps are more and more ephemeral in nature, existing only for a short time (event, or intensive field effort), or are being geared toward a "citizen science" model.  The existing license frameworks just don't make sense in that context.

Some sort of clear cut "pay-to-play" license model would greatly expand the usability of the runtime SDKs outside the scope of traditional GIS activities.

**Comment by mdcooper** [comment]
Yes, would love a way to license an app without requiring "named users" and without forcing us to purchase license packs when we already have both Portal and AGOL org subscription.  Our use case isn't for a public app, but simply for an org that wants to use offline editing in an app and doesn't want anything to do with portal or AGOL.  We have AGOL and Portal, but it just seems unfortunate to force the dependency on them if you don't want to use anything that they provide.

**Comment by EuanCameron** [comment]
When using an embedded deployment license the process is the same for all runtime SDKs.    You must record the number of deployments sold and purchase the appropriate number of deployment licenses from Esri customer service.

https://developers.arcgis.com/en/arcgis-runtime/licensing/

**Comment by madmapper** [comment]
Thanks for the update - I have to wonder though if Esri would ever consider a license for the SDK (or perhaps a subset of the standard-level SDK functionality) that was offered under a licensing structure more compatible with consumer apps.

The realities of the consumer app store economies are such that offering an app at a price point compatible with a $200 runtime license code or yearly portal fee is unrealistic for almost all circumstances except B2B and enterprise.

Perhaps supporting consumer apps doesn't align with whatever vision Esri has for the Runtime SDKs, but I certainly think there is room for *someone* to enter the space with an SDK that supports offline collection and map display.

**Comment by eric906** [comment]
@margitbomholt , at this time, there is no license model for this. It is a great question, for sure, but we have not decided yet on how to support this deployment use case, so at this time it isn't supported.

Esri is focused more on the Enterprise deployment model of Standard-level Runtime apps, regardless of organizational accounts or just straight purchased deployments. At least today. Consumer-focused apps are generally going to be those Basic-level Runtime apps you might see in the stores right now, at least that is the recommendation from Esri today. Make sense?
